### PR TITLE
adding third party lib for Glob pattern matching which supports /**/ style subdirectory matching

### LINKED ISSUE #201: does promtail support Subdirectory
does promtail support Subdirectory? if it does, how to config it.

### PR DESCRIPTION
**Pull request by slim-bean**
Fixes #201 

Also includes some more thorough testing of the promtail client

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -79,3 +79,7 @@
 [prune]
   go-tests = true
   unused-packages = true
+
+[[constraint]]
+  name = "github.com/bmatcuk/doublestar"
+  version = "1.1.1"
```
Comment (tomwilkie) [inline Gopkg.toml]: Is the constraint necessary?  Can't we use the latest and greatest?  In general, every entry in this file needs a justification.

**Comment by tomwilkie** [comment]
(ps if you say "fixes" instead of "addresses", github will close the issue for you when this is merged)

**Comment by robbieaverill** [comment]
Thanks for adding this! Just wondering, can we update the promtail config files to use this pattern so it's clear when reading them that they will support subdirectories? https://github.com/grafana/loki/blob/778c0b6/cmd/promtail/promtail-local-config.yaml#L18

e.g. `__path__: /var/log/**/*.log` ?

**Comment by azawawi** [comment]
I encountered the same issue while trying out the docker installation for the first time. I think @robbieaverill suggestion to add `__path__: /var/log/**/*.log` to https://github.com/grafana/loki/blob/778c0b6/cmd/promtail/promtail-local-config.yaml#L18 is correct.

**Comment by willzhang** [comment]
have been support!

**Pull request merged**